dev/com.ibm.ws.microprofile.config.1.2/src/com/ibm/ws/microprofile/config12/converters/MethodFunction.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config12.converters;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.function.Function;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+import com.ibm.ws.ffdc.annotation.FFDCIgnore;
+import com.ibm.ws.microprofile.config.interfaces.ConversionException;
+
+public class MethodFunction<X> implements Function<String, X> {
+    private final Method method;
+
+    protected MethodFunction(Method method) {
+        this.method = method;
+    }
+
+    /** {@inheritDoc} */
+    @SuppressWarnings("unchecked")
+    @Override
+    public X apply(String value) {
+        X converted = null;
+        if (value != null) { //if the value is null then we always return null
+            try {
+                converted = (X) method.invoke(null, value);
+            } catch (InvocationTargetException e) {
+                Throwable cause = e.getCause();
+                if (cause instanceof IllegalArgumentException) {
+                    throw (IllegalArgumentException) cause;
+                } else {
+                    throw new ConversionException(cause);
+                }
+            } catch (IllegalAccessException e) {
+                throw new ConversionException(e);
+            }
+        }
+        return converted;
+    }
```

Comment:
Should this have `@FFDCIgnore` since we're throwing the exception up to the user anyway?

An exception here presumably means they had a method or constructor to create an object from a String, but calling it threw an exception?

Reply:
We should probably ignore `InvocationTargetException` but still FFDC for `IllegalAccessException`

Reply:
done